.github/workflows/e2e-tests.yml

```diff
@@ -40,4 +45,43 @@ jobs:
       - name: Stop containers
         working-directory: extensions/ql-vscode/test/e2e
         if: always()
-        run: docker-compose -f "docker-compose.yml" down -v
\ No newline at end of file
+        run: docker-compose -f "docker-compose.yml" down -v
+  report-failure:
+    name: Report failure on the default branch
+    runs-on: ubuntu-latest
+    needs: [e2e-test]
+    if: failure() && github.ref == 'refs/heads/main'
+    permissions:
+      contents: read
+      issues: write
+    env:
+      GH_TOKEN: ${{ github.token }}
+    steps:
+      - name: Create GitHub issue
+        run: |
+          # Set -eu so that we fail if the gh command fails.
+          set -eu
+
+          # Try to find an existing open issue if there is one
+          ISSUE="$(gh issue list --repo "$GITHUB_REPOSITORY" --label "e2e-test-failure" --state "open" --limit 1 --json number -q '.[0].number')"
```

Comment:
Note you'll need to create the `e2e-test-failure` label

Reply:
Need to also make sure that that label exists, otherwise it'll fail